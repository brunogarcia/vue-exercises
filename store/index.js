
import Vuex from 'vuex';

const store = () => {
    return new Vuex.Store({
        state: {
            missionText: '',
            hustonOpen: false,
        },
        getters: {
            hustonText(state) {
                return state.missionText;
            },
            hustonOpen(state) {
                return state.hustonOpen;
            }
        },
        mutations: {
            SET_HUSTON_TEXT(state, text) {
                state.missionText = text;
            },
            CALL_TO_HUSTON(state, val) {
                state.hustonOpen = !!val;
            },
        },
        actions: {
            updateHustonText({ commit }, text) {
                commit('SET_HUSTON_TEXT', text);
            },
            callToHuston({ commit }, val) {
                commit('CALL_TO_HUSTON', val);
            }
        }
    });
};

export default store;
