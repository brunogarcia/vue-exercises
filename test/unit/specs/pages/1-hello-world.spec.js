import {
    mount,
    createLocalVue
} from '@vue/test-utils';
import Vuex from 'vuex';
import Vuetify from 'vuetify';
import Hello from '~/pages/exercise-1-hello-world/index.vue';
import store from '../../../fixtures/store';

describe('Home /', () => {
    let localVue;

    beforeEach(() => {
        localVue = createLocalVue();
        localVue.use(Vuetify);
        localVue.use(Vuex);
    });

    it('renders Hello world page', () => {
        const wrapper = mount(Hello, {
            localVue: localVue,
            store: store(),
            stubs: ['nuxt-link', 'the-satelite']
        });
        expect(wrapper.isVueInstance()).toBeTruthy();
        expect(wrapper.vm.$el).toMatchSnapshot();
    });

    afterAll(() => {
        localVue.destroy();
    });
});
