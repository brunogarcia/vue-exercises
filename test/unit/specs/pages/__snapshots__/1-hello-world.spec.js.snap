// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home / renders Hello world page 1`] = `
<section
  class="container"
>
  <!---->
   
  <div
    class="satelite-wrapper prel"
  >
    <the-satelite-stub />
  </div>
   
  <div
    class="title"
  >
    Hello, ...world?
  </div>
   
  <div
    class="body-1"
  />
</section>
`;
